### DISCUSSION TITLE
Need help with json.patch built-in

### DISCUSSION THREAD

**Discussion by cmorinupgrade** [open-policy-agent/community#361] [OPA and Rego] [answered]
Hi folks,

I am having a hard time getting the json.patch built-in to work with my rule.

Let's say I got this object:
```
required_approvals := {
    "approvals": [{
        "users": [],
        "files": [],
    }]
}
```

As part of a comprehension, I'd like to dynamically patch that object. For the sake of simplicity, I'm trying to go with a very simple example to get to a working solution. However, I did not have much success so far.

```
patch := {
    "users": [
        "user1",
        "user2",
    ],
    "files": [
        "file1",
        "file2",
     ],
}
```

then I'm trying with this:
```
required_approvals = json.patch(required_approvals, [{ "op": "add", "path": "/approvals/0", "value": patch } ])
```

However, when printing `required_approvals` after the `patch`, I got this, which suggests it's not working as I am thinking it should work:
`{"approvals": [{"files": [], "users": []}]}`

Looking at the output of `json.patch`, it seems to work. However, it does not seem to actually change the initial `required_approvals` object I defined, which is what I would like. I tried doing `required_approvals = output`, but nothing changes.

I do not get any error messages.

Am I missing something?

Thank you in advance for help.

**Comment by anderseknert** [comment, marked as answer]
Hi there! 👋 Variables in Rego are immutable, so you can't change the value of `required_approvals`. You can however assign the result of the patch operation to a new var. I'm surprised you did not see an error message about that. If you can share your policy in full, I'd be happy to take a look. 

See example of working policy here: https://play.openpolicyagent.org/p/vQQQopNORc
